src/internal.cc

```diff
@@ -132,7 +132,7 @@ internalRemoteUri(const char *host, unsigned short port, const char *dir, const
 char *
 internalLocalUri(const char *dir, const SBuf &name)
 {
-    return internalRemoteUri(getMyHostname(),
+    return internalRemoteUri(false, getMyHostname(),
```

Comment:
Why always `false`? The call to getMyPort() may return a secure port, right?

internalLocalUri() itself is a broken/outdated concept because the local URL depends on the (missing) caller context -- there may be no single correct local URL for multiport Squid instances. In the extreme (and possibly currently impossible) case, the right scheme might even be `ftp://`! You may not want to address this problem in this PR at all, but please keep it in mind while you polish this PR.

Reply:
Yes...hard coded to false is just to keep this part as its old behavior...

Because not like other places the tls info is just at hand, here I'm afraid changing that will drag a lot things in and I may make this PR worse.

The PR is here because we need to secure the traffic between our cross country peers, there are interceptions/filters in the boundary, you know. So there is no workaround for us like opening a non-encrypted port for peers to talk, even by locking down both sides' IPs, as packets flow, they will be dropped.

Actually I am not quite sure about the usage of internalLocalUri(), but from its implementation, I think it is to provide loopback access to the squid instance itself. At least we can afford opening a non-encrypted http port as the first server port on lookback interface. So I it is not a must have in this PR. I feel I'd better to make this PR small to do at least one thing right...

Reply:
If you add the sometimes wrong `false` parameter, please do add an XXX source code comment to explain why you are adding one sleeper bug while fixing another. For example:

```suggestion
    // XXX: getMy*() may return https_port info, but we force http URIs
    // because we have not checked whether the callers can handle https.
    const bool secure = false;
    return internalRemoteUri(secure, getMyHostname(),
```